protocols/kad/src/lib.rs

```diff
@@ -37,6 +37,7 @@
 //       be useful later for record store
 #![allow(dead_code)]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
+#![allow(deprecated)]
 
 pub mod handler;
 pub mod kbucket;
```

Comment:
(Can't select all four modules but applies to all of them)

Instead of deprecating individual types, we can safe ourselves some typing and this:

```rust
mod handler_priv;

#[deprecated(note = "The `handler` module will be made private in the future and should not be depended on.")]
pub mod handler {
	pub use super::handler_priv::*;
}
```

Internally, we want to import everything from the `_priv` modules to avoid triggering deprecation warnings.

Any type that appears in the actual public API should be re-exported from the root module using `pub use`.